backend/courses/management/commands/fill_parent_courses.py

```diff
@@ -0,0 +1,81 @@
+from textwrap import dedent
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import F, Max, OuterRef, Subquery, Value
+
+from courses.models import Course
+from courses.util import get_semesters
+
+
+def fill_parent_courses(semesters: list[str], verbose=False):
+    """
+    Attemps to fill empty `parent_course` fields for all courses in the specified semesters.
+
+    :param semesters: Semesters for which you want to fill `parent_course` fields.
+    :param verbose: Whether to print status/progress updates.
+    """
+    if verbose:
+        print(f"Attempting to fill empty `parent_course` fields for semesters {semesters}.")
+    filled_links = 0
+    for i, semester in enumerate(sorted(semesters)):
+        if verbose:
+            print(f"Processing semester {semester} ({i+1}/{len(semesters)})...")
+        with transaction.atomic():
+            courses = list(Course.objects.filter(semester=semester, parent_course__isnull=True))
+            full_codes = {course.full_code for course in courses}
+            parent_candidates = (
+                Course.objects.filter(full_code__in=full_codes, semester__lt=semester)
```

Comment:
Should we allow for small variations in full code (e.g. prepending or appending 0)?

Reply:
I think this makes sense since this is generally a heuristic used by Wharton, CIS and others

Reply:
How are we planning on using the crosswalk?

Reply:
The NGSS crosswalk? That's loaded at the end of this script.

For now let's just stay on the extra careful side and only link 3-4 digit codes if it's included in the crosswalk or we manually add a `parent_course` link. We really don't want to continue to be plagued by the topic blowup problem.